xdsl/dialects/omp.py

```diff
@@ -520,7 +522,7 @@ class WsLoopOp(BlockArgOpenMPOperation):
         ArrayAttr.constr(RangeOf(base(SymbolRefAttr), length=REDUCTION_COUNT))
     )
     reduction_mod = opt_prop_def(ReductionModifierAttr)
-    reduction_byref = opt_prop_def(DenseIntOrFPElementsAttr[i1])
+    reduction_byref = opt_prop_def(DenseArrayBase[i1])
```

Comment:
Are these part of a different change? As an aside, it is very confusing to me that this typechecks, `i1` is an instantiation of `IntegerType` so how comes it can be a generic type parameter?

Reply:
No problem